webapp-django/crashstats/signature/views.py

```diff
@@ -437,33 +438,13 @@ def signature_summary(request, params):
 
     facets = search_results['facets']
 
-    # We need to make a separate query so that we can show all versions and
-    # not just the one asked for.
-    params_copy = {
-        'signature': params['signature'],
-        '_aggs.product.version': ['_cardinality.install_time'],
-    }
-
-    try:
-        product_results = api.get(**params_copy)
-    except BadArgumentError as e:
-        # We need to return the error message in some HTML form for jQuery
-        # to pick it up.
-        return http.HttpResponseBadRequest(render_exception(e))
-
-    if 'product' in product_results['facets']:
-        facets['product'] = product_results['facets']['product']
-    else:
-        facets['product'] = []
```

Comment:
Previously, this would do a separate API get for product results and then stick them into the context. The new version doesn't do that--so there are no product results in the facet as near as I can tell.

How does product information get into the signature report page via `context['query']`?

Reply:
I added `params['_aggs.product.version']`. This makes the first API call add the product results. They are in `search_results['facets']['product']`.  `context['query']` is set to be `search_results`, so that's how they are available.

The facets dictionary is obtained from `search_results` at line 438. Previously, `facets['product']` would be overwritten at line 455 by `product_results['facets']['product']`.

Reply:
Ahhh... That makes sense!